engine/gui/widgets.py

```diff
@@ -89,10 +91,83 @@ def create(
                 None,
             )
 
-        return GUISpec(
+        gui_spec = GUISpec(
             assets=[Asset(**a) for a in spec.get("assets", [])],
             buttons=buttons,
             images=[Image(**b) for b in spec.get("images", [])],
             cancel_action=cancel_action,
             initial_selected_button=selected_button,
         )
+
+        cls.validate(gui_spec)
```

Comment:
I'm surprised this works, probably a quirk with how Python handles classes. Usually you call an instance method like this:
```python
gui_spec.validate()
```

Reply:
Yeah, I think failed to merge that change. I'll change it again.